dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js

```diff
@@ -2,1174 +2,1088 @@ dojo.provide("dotcms.dijit.image.ImageEditor");
-
-
-    },
-    setCropHeightFromWidth : function(){
-
-        var cw = dojo.attr(this.iframe.dojo.byId("cropWidth"),"value");
-        var coord = dojo.coords(this.iframe.dojo.byId("me"));
-        var width = coord.w;
-        var height = coord.h;
-
-        dojo.attr(this.iframe.dojo.byId("cropHeight"),"value", Math.floor(height * (cw/width) ));
-    },
-
-    setCropWidthFromHeight: function(){
-
-        var ch = dojo.attr(this.iframe.dojo.byId("cropHeight"),"value");
-        var coord = dojo.coords(this.iframe.dojo.byId("me"));
-        var width = coord.w;
-        var height = coord.h;
-
-        dojo.attr(this.iframe.dojo.byId("cropWidth"),"value", Math.floor(width * (ch/height) ));
-    },
-
-
-
-
-
-    // turns cropping on
-    _cropOn: function(){
-
-        dojo.style(this.iframe.dojo.byId("cropBtn"), "border", "red 1px solid");
-        if(this.crop && this.crop != undefined){
-            return;
-        }
-
-
-        this.crop = new dotcms.dijit.image.Crop({
-            hoverable:true
-        }, this.iframe.dojo.byId("me"));
-        this.crop.showCropBox();
-
-    },
-    _cropOff: function(){
-
-        if(!this.crop || this.crop ==undefined){
-            return;
-        }
-        dojo.style(this.iframe.dojo.byId("cropBtn"), "border", "red 0px dotted");
-
-        var img = this.iframe.dojo.byId("me");
-
-        this.crop.destroy();
-        this.crop=null;
-        this.iframe.dojo.byId("imageViewPort").appendChild(img);
-    },
-
-    doCrop: function(){
-        this.iframe.dojo.byId("cropBtn").blur();
-        dojo.attr(this.iframe.dijit.byId('cropBtn'), "disabled", true);
-
-        var showImage = this.iframe.dojo.byId("me");
-        var baseImage = this.iframe.dojo.byId("baseImage");
-
-        pc = dojo.coords(this.crop.picker);
-        vp = dojo.coords(this.iframe.dojo.byId("imageViewPort"));
-        sTop = this.iframe.dojo.byId("imageViewPort").scrollTop;
-        sLeft = this.iframe.dojo.byId("imageViewPort").scrollLeft;
-
-        var sw = parseInt(this.iframe.dojo.byId("displayImageWidth").value);
-        var sh = parseInt(this.iframe.dojo.byId("displayImageHeight").value);       
+dojo.declare("dotcms.dijit.image.ImageEditor", dijit._Widget, {
+	// cssUrl:'/html/js/dotcms/dijit/image/image-tools.css',
+	imageToolJsp: "/html/js/dotcms/dijit/image/image_tool.jsp",
+	baseFilterUrl: "/contentAsset/image",
+	// changes to show the value in the toolbar
+	currentUrl: "/contentAsset/image",
+	ajaxUrl: "/servlet/dotImageToolAjax",
+
+	resizeFilter: false,
+	zoomValue: 0,
+	inode: "0",
+	tempId: null,
+	inited: false,
+	painting: false,
+	thumbnailWidth: 500,
+	thumbnailHeight: 250,
+	thumbDivId: "",
+	fieldName: "fileAsset",
+	saveAsIncrement: 1,
+	editImageText: "Edit Image",
+	binaryFieldId: "",
+	fieldContentletId: "",
+	saveAsFileName: "",
+	compression: "none",
+	compressionValue: 65,
+	fileSize: 0,
+	execute: null,
+	activeEditor: undefined,
```

Comment:
Unfortunately, I messed up with the formatting so I'll comment on each change made in this file.

Added two props: `execute` and `activeEditor`. The `execute()` function is used to call `createImageWindow` from  the WYSIWYG editor.

`activeEditor` is the instance of the active editor so we can insert the image in the current editor.